.github/workflows/unit-test.yml

```diff
@@ -0,0 +1,169 @@
+        required: false
+        type: string
+        default: ''
+      install_pennylane_lightning_master:
+        description: Indicate if PennyLane-Lightning should be installed from the master branch
+        required: false
+        type: boolean
+        default: true
+      pytest_test_directory:
+        description: The directory where the PennyLane tests are that should be run by PyTest
+        required: false
+        type: string
+        default: tests
+      pytest_coverage_flags:
+        description: Coverage flags for PyTest
+        required: false
+        type: string
+        default: ''
+      pytest_markers:
+        description: Custom mark string to pass to PyTest
+        required: false
+        type: string
+        default: ''
+      pytest_additional_args:
+        description: Additional arguments to pass to PyTest
+        required: false
+        type: string
+        default: ''
+      additional_pip_packages:
+        description: Additional packages to install. Values will be passed to pip install {value}
+        required: false
+        type: string
+        default: ''
+
+jobs:
+  test:
+    name: ${{ inputs.job_name }}
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ inputs.branch }}
+          fetch-depth: ${{ inputs.checkout_fetch_depth }}
+
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '${{ inputs.python_version }}'
+
+      - name: Upgrade PIP
+        run: pip install --upgrade pip && pip install wheel --upgrade
+
+      - name: Install PennyLane dependencies
+        run: |
+          pip install -r requirements-ci.txt --upgrade
+          pip install -r requirements-dev.txt --upgrade
+
+      - name: Install PyTorch
+        if: inputs.install_pytorch == true
+        env:
+          TORCH_VERSION: ${{ inputs.pytorch_version != '' && format('=={0}', inputs.pytorch_version) || '' }}
+        run: pip install "torch${{ env.TORCH_VERSION }}" -f https://download.pytorch.org/whl/torch_stable.html
+
+      - name: Install TensorFlow
+        if: inputs.install_tensorflow == true
+        env:
+          TF_VERSION: ${{ inputs.tensorflow_version != '' && format('~={0}', inputs.tensorflow_version) || '' }}
+        run: pip install "tensorflow${{ env.TF_VERSION }}" "keras${{ env.TF_VERSION }}"
+
+      - name: Install JAX
+        if: inputs.install_jax == true
+        env:
+          JAX_VERSION: ${{ inputs.jax_version != '' && format('=={0}', inputs.jax_version) || '' }}
+        run: pip install "jax${{ env.JAX_VERSION}}" "jaxlib${{ env.JAX_VERSION }}"
+
+      - name: Install additional PIP packages
+        if: inputs.additional_pip_packages != ''
+        env:
+          PIP_PACKAGES: ${{ inputs.additional_pip_packages }}
+        run: |
+          if [[ ! "$PIP_PACKAGES" =~ ^[a-zA-z0-9~=\-_:\/@\+\.]+ ]]; then
```

Comment:
did you encounter something that made you want to add this validation step? not opposed, regex is just scary and I would have just blindly trusted the user in this case lol

Reply:
Most definitely can take the user input verbatim ... but I wanted to ensure it is the name(s) of pip packages, that being said yeah it likely is overkill :sweat_smile: , let me make it simpler